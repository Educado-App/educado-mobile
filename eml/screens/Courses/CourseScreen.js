import React from 'react'
import { View, StyleSheet } from 'react-native'
import CourseTitleIcon from '../../components/courses/courseHeader/CourseTitleIcon'
import CourseProgress from '../../components/courses/courseHeader/CourseProgress'
import CourseMenu from '../../components/courses/courseHeader/CourseMenu'

export default function CourseScreen() {
  const color = '#006622'
  const type = 'material-community'
  const name = 'cash'

  return (
    <View style={styles.container}>
      <View>
        <CourseMenu />
      </View>
      <View>
        <CourseTitleIcon
          color={color}
          name={name}
          type={type}
          title={'Personal Finance'}
        ></CourseTitleIcon>
      </View>
      <CourseProgress fracTop={45} fracBot={100}></CourseProgress>
    </View>
  )
}

const styles = StyleSheet.create({})
