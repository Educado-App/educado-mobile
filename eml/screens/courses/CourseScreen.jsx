import {useNavigation} from '@react-navigation/native';
import React, {useEffect, useState} from 'react';
import {Image, Pressable, RefreshControl, ScrollView, View} from 'react-native';
import Text from '../../components/general/Text';
import * as StorageService from '../../services/StorageService';
import CourseCard from '../../components/courses/courseCard/CourseCard';
import BaseScreen from '../../components/general/BaseScreen';
import IconHeader from '../../components/general/IconHeader';
import {shouldUpdate} from '../../services/utilityFunctions';
import ToastNotification from '../../components/general/ToastNotification';
import LoadingScreen from '../../components/loading/Loading';
import NetworkStatusObserver from '../../hooks/NetworkStatusObserver';
import AsyncStorage from '@react-native-async-storage/async-storage';
import errorSwitch from '../../components/general/errorSwitch';
import ShowAlert from '../../components/general/ShowAlert';

/**
 * Course screen component that displays a list of courses.
 * @component
 * @returns {JSX.Element} The course screen component.
 */
export default function CourseScreen() {
	const [courses, setCourses] = useState([]);
	const [courseLoaded, setCourseLoaded] = useState(false);
	const [refreshing, setRefreshing] = useState(false);
	const [isOnline, setIsOnline] = useState(false);
	const [loading, setLoading] = useState(true);
	const navigation = useNavigation();

	/**
    * Asynchronous function that loads the courses from storage and updates the state.
    * @returns {void}
    */
	async function loadCourses() {
		const courseData = await StorageService.getSubCourseList();
		if (shouldUpdate(courses, courseData)) {
			if (courseData.length !== 0 && Array.isArray(courseData)) {
				setCourses(courseData);
				setCourseLoaded(true);
			}
			else {
				setCourses([]);
				setCourseLoaded(false);
			}
		}
		setLoading(false);
	}

	// When refreshing the loadCourses function is called
	const onRefresh = () => {
		setRefreshing(true);
		loadCourses();
		setRefreshing(false);
	};

	useEffect(() => {
		// this makes sure loadCourses is called when the screen is focused
		return navigation.addListener('focus', () => {
			loadCourses();
		});
	}, [navigation]);

	useEffect(() => {
		const logged = async () => {
			const loggedIn = await AsyncStorage.getItem('loggedIn');
			if (loggedIn) {
				setTimeout(async () => {
					ToastNotification('success', 'Logado!');
					await AsyncStorage.removeItem('loggedIn');
				}, 1000);
			}
		};
		try {
			logged();
		} catch (e) {
			ShowAlert(errorSwitch(e));
		}
	}, []);

	return (
		loading ? <LoadingScreen /> :
			<>
				<NetworkStatusObserver setIsOnline={setIsOnline}/>
				<BaseScreen>
					{/** Checks if the course(s) has been loaded
           * If it has, it will render and map the courses
           * If not, it will render a message saying that there are no active courses (in portugese)
           */}
					{courseLoaded ?
						<View height="100%">
							<IconHeader
								title={'Bem Vindo!'}
								description={'Aqui você encontra todos os cursos em que você está inscrito!'}/>
							<ScrollView showsVerticalScrollIndicator={false}
								refreshControl={<RefreshControl refreshing={refreshing} onRefresh={onRefresh}/>}>
								{courses.map((course, index) => (
									<CourseCard key={index} course={course} isOnline={isOnline}></CourseCard>
								)
								)}
							</ScrollView>
						</View>
						:
						<View className="bg-secondary justify-center items-center ">
							<View className="pt-24 pb-16">
								<Image source={require('../../assets/images/logo.png')} className=" justify-center items-center"/>
							</View>
							<View className=" justify-center items-center py-10 gap-10 ">
								<View className=" justify-center items-center w-full h-auto  px-10">
									{/* No active courses */}
									<Image source={require('../../assets/images/no-courses.png')}/>
									<Text
										className=" leading-[29.26] text-projectBlack pb-4 pt-4 font-sans-bold text-subheading text-center ">Comece
                      agora</Text>
									<Text className=" text-projectBlack font-montserrat text-center text-body "> Você ainda não se
                      increveu em nenhum curso. Acesse a página Explore e use a busca para encontrar cursos do seu
                      intresse.</Text>
									{/*You haven't signed up for any course yet. Access the Explore page and use the search to find courses that interest you.*/}
								</View>
								<View>
									<Pressable
										testID={'exploreButton'}
										className=" rounded-r-8 rounded-md bg-primary justify-center items-center py-4 w-full h-auto px-20 "
										onPress={() => navigation.navigate('Explorar')}>
										{/* Click to explore courses */}
										<Text className=" text-projectWhite font-sans-bold text-center text-body "> Explorar cursos</Text>
									</Pressable>
								</View>
							</View>
						</View>}
				</BaseScreen>
			</>
	);
}
