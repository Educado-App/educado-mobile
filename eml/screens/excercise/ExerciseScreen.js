import { StatusBar } from "expo-status-bar";
import React, { useEffect, useRef, useState } from "react";
import { Alert, StyleSheet, View, Text } from "react-native";
import LeaveButton from "../../components/exercise/LeaveButton";
import ExerciseButtons from "../../components/exercise/ExerciseButtons";
import { useNavigation, useRoute } from "@react-navigation/native";
import { Video } from "expo-av";
import * as StorageService from "../../services/StorageService";
import CustomProgressBar from "../../components/exercise/Progressbar";
import dummyExerciseData from './dummyExerciseData.json';



export default function ExerciseScreen() {
  const navigation = useNavigation();
  const route = useRoute();
  
  const [hasData, setHasData] = useState(false);
  const [signal, setSignal] = useState([]);
  const [exerciseData, setExerciseData] = useState({});

  /*async function getExercise() {
    const exercise = await StorageService.getNextExercise(sectionId);

    if (exercise !== null) {
      console.log(exercise);

      if (exercise === true) {
        navigation.navigate("SectionComplete", {
          courseId: courseId,
          sectionId: sectionId,
        });
      } else {
        setExerciseData(exercise);
      }
    } else {
      navigation.navigate("ErrorScreen");
    }
  }*/

  async function getExercise() {
    // Instead of calling StorageService.getNextExercise, use the dummy data
    const exercise = dummyExerciseData;
  
    if (exercise !== null) {
      console.log(exercise);
  
     /* const courseId = exercise.courseId || "defaultCourseId";
      const sectionId = exercise.sectionId || "defaultSectionId";*/
  
      if (exercise === true) {
       /* navigation.navigate("SectionComplete", {
          courseId: courseId,
          sectionId: sectionId,
        });*/
      } else {
        setExerciseData(exercise);
      }
    } else {
      navigation.navigate("ErrorScreen");
    }
  }
  
  

  /*async function isSectionComplete(courseId, sectionId) {
    const course = await StorageService.getCourseById(courseId);
    const sections = course.sections;

    for (let i = 0; i < sections.length; i++) {
      console.log("Section is complete: ", sections[i].isComplete);
      if (sections[i].id === sectionId) {
        return sections[i].isComplete;
      }
    }
  }*/

  useEffect(() => {
    getExercise().then(() => {
      setHasData(true);
    });
  }, [route.params]);

  const video = useRef(0);

  useEffect(() => {
    if (video !== undefined) {
      if (signal === 0) {
        try {
          video.current.pauseAsync();
        } catch (e) {
          console.log("Something went wrong :" + e);
        }
      }
      setSignal(1);
    }
  }, [signal]);

  return (
    <View style={styles.container} className="bg-babyBlue">
      <View style={{ flex: 1 }}>
        <View className="pt-8 flex-row w-screen">
          <View className="">
            <LeaveButton
              //navigationPlace={"Course"}
              //courseId={courseId}
            ></LeaveButton>
          </View>
          <View
            style={{
              left: "80%",
              justifyContent: "center",
              alignItems: "center",
              flexDirection: "column",
            }}
          ></View>
        </View>
        <View className="items-center">
          <CustomProgressBar progress={0.5 / 1}></CustomProgressBar>
        </View>
      </View>
      <View style={{ flex: 2, width: "100%" }}>
        {exerciseData === undefined ? (
          // No Data
          <Text> Sem dados</Text>
        ) : (
          <Video
            source={{ uri: exerciseData.content }}
            rate={1.0}
            volume={1.0}
            isMuted={false}
            resizeMode="cover"
            shouldPlay
            useNativeControls
            ref={video}
            style={styles.backgroundVideo}
          />
        )}
      </View>
      <View style={{ flex: 3 }}>
        {exerciseData === undefined ? (
          //No data
          <Text> Sem dados</Text>
        ) : (
          <ExerciseButtons
            answers={exerciseData.answers}
            exerciseId={exerciseData.id}
            courseId={exerciseData.courseId}
            sectionId={exerciseData.sectionId}
            setSignal={setSignal}
            hasData={hasData}
          ></ExerciseButtons>
        )}
      </View>
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
  },
  row: {
    flexDirection: "row",
  },
  textStyle: {
    zIndex: 100,
    fontSize: 100,
  },
  buttonShadow: {
    shadowOpacity: 0.4,
    shadowRadius: 10,
    shadowOffset: { width: 0, height: 0 },
  },
  backgroundVideo: {
    height: "100%",
  },
});
