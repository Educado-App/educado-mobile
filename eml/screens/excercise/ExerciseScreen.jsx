import React, { useState } from 'react';
import { ScrollView, View, TouchableOpacity } from 'react-native';
import Text from '../../components/general/Text';
import { RadioButton } from 'react-native-paper';
import ExerciseInfo from '../../components/exercise/ExerciseInfo';
import { Icon } from '@rneui/themed';
import { SafeAreaView } from 'react-native-safe-area-context';
import PopUp from '../../components/gamification/PopUp';
import { StatusBar } from 'expo-status-bar';
import PropTypes from 'prop-types';
import { completeComponent, handleLastComponent } from '../../services/utilityFunctions';
import { useNavigation } from '@react-navigation/native';

/* 
Description: 	This screen is displayed when the student is doing an exercise.
				It displays the question and the answers, and the student can select one answer.
				When the student presses the confirm button, the answer is checked and the student is given feedback.
				When the student presses the continue button, the next component is displayed.
				The student can only continue if an answer is selected.
				The student is given 10 points when the answer is correct in the first try, 
				otherwise the student gets 5 points when the answer is correct.
				The student gets 0 points when the answer is incorrect or they have completed the exercise before.
Dependencies: 	CompSwipeScreen, the screen which contains all the components in the section
Props:			- exerciseObject: The exercise object, which contains the question and the answers
				- sectionObject: The section object, which contains the section title	
				- courseObject: The course object, which contains the course title
				- onContinue: A function that is called when the student presses the continue button, 
				when the exercise is completed and it is the last component in the section, the student is taken to the section complete screen
*/

export default function ExerciseScreen({ exerciseObject, sectionObject, courseObject, onContinue }) {
	const tailwindConfig = require('../../tailwind.config.js');
	const projectColors = tailwindConfig.theme.colors;
	const navigation = useNavigation();

	const [selectedAnswer, setSelectedAnswer] = useState(null);
	const [buttonClassName, setButtonClassName] = useState('');
	const [showFeedback, setShowFeedback] = useState(false);
	const [buttonText, setButtonText] = useState('Confirmar Resposta'); 
	const [isPopUpVisible, setIsPopUpVisible] = useState(false); 
	const [isCorrectAnswer, setIsCorrectAnswer] = useState(false);
	const [points, setPoints] = useState(10);

	const handleAnswerSelect = (answerIndex) => {
		setSelectedAnswer(answerIndex);
	};
  
	async function handleReviewAnswer(selectedAnswer) {
		const continueText = 'Continuar';

		setIsCorrectAnswer(selectedAnswer);

		setButtonClassName(
			`bg-project${selectedAnswer ? 'Green' : 'Red'}`
		);

		setShowFeedback(true);
		setButtonText(continueText);
		if (buttonText !== continueText) {
			const obj = await completeComponent(exerciseObject, courseObject.courseId, selectedAnswer);
			setPoints(obj.points);
			setIsPopUpVisible(true);
		} else {
			setIsPopUpVisible(false);
			if (onContinue()) {
				handleLastComponent(exerciseObject, courseObject, navigation);
			}
		}
	}
  

	return (
		<SafeAreaView className="h-full bg-secondary">

			<View className='items-center'>
				<Text testID='exerciseQuestion'
					className='pt-20 pb-10 text-center text-body font-sans-bold text-projectBlack w-11/12'>
					{exerciseObject.question}
				</Text>

				<View className={`${buttonClassName} items-center justify-center h-96 w-full`}>
					<ScrollView className="py-2">
						{/* Map through the answers and render each one */}
						{exerciseObject.answers.map((answer, index) => (
							<View
								key={index}
								className='flex-row w-96 pb-6 pl-2'
							>
								<View>
									<RadioButton.Android
										disabled={showFeedback}
										value={index}
										status={
											selectedAnswer === index ? 'checked' : 'unchecked'
										}
										onPress={() => handleAnswerSelect(index)}
										color={projectColors.primary}
										uncheckedColor={projectColors.primary}
									/>
								</View>

								<View>
									<TouchableOpacity onPress={() => handleAnswerSelect(index)} disabled={showFeedback}>
										<Text className='pt-2 pb-1 w-72 font-montserrat text-body text-projectBlack'>{answer.text}</Text>
									</TouchableOpacity>

									{showFeedback ? (
										<View className={`flex-row pb-2 w-fit rounded-medium ${answer.correct ? 'bg-projectGreen' : 'bg-projectRed'}`}>
											<View className='pl-2 pt-1'>
												<View className='pt-1.5'>
													{answer.correct === true ? (
														<Icon
															size={10}
															name='check'
															type='material'
															color={projectColors.success}
														/>
													) : (
														<Icon
															size={10}
															name='close'
															type='material'
															color={projectColors.error}
														/>
													)}
												</View>
											</View>
											<Text className={`w-72 pl-1 pt-2 pr-2 text-caption-medium ${answer.correct ? 'text-success' : 'text-error'}`}>{answer.feedback}</Text>
										</View>
									) : null}
								</View>
							</View>
						))}
					</ScrollView>
				</View>
				<View className='px-6 pt-10 w-screen'>
					<TouchableOpacity
						disabled={selectedAnswer === null ? true : false}
						className={`${selectedAnswer !== null ? 'opacity-100' : 'opacity-30'} bg-primary px-10 py-4 rounded-medium`}
						onPress={() => handleReviewAnswer(exerciseObject.answers[selectedAnswer].correct)}
					>
						<Text className='text-center font-sans-bold text-body text-projectWhite'>{buttonText}</Text>
					</TouchableOpacity>
				</View>
			</View>
    

			{isPopUpVisible ? (
				<PopUp pointAmount={points} isCorrectAnswer={isCorrectAnswer} />
			) : null}

			{<ExerciseInfo courseTitle={courseObject.title} sectionTitle={sectionObject.title} />}
			<StatusBar style='auto' />
		</SafeAreaView>
	);
}

ExerciseScreen.propTypes = {
	exerciseObject: PropTypes.object,
	sectionObject: PropTypes.object,
	courseObject: PropTypes.object,
	onContinue: PropTypes.func,
};

/*
async function getExercise() {
  const exercise = await StorageService.getNextExercise(sectionId);

  if (exercise !== null) {
    console.log(exercise);

    if (exercise === true) {
      navigation.navigate("SectionComplete", {
        courseId: courseId,
        sectionId: sectionId,
      });
    } else {
      setExerciseData(exercise);
    }
  } else {
    navigation.navigate("ErrorScreen");
  }
}*/

/*async function isSectionComplete(courseId, sectionId) {
const course = await StorageService.getCourseById(courseId);
const sections = course.sections;

for (let i = 0; i < sections.length; i++) {
  console.log("Section is complete: ", sections[i].isComplete);
  if (sections[i].id === sectionId) {
    return sections[i].isComplete;
  }
}
}

async function getExercise() {
  // Instead of calling StorageService.getNextExercise, use the dummy data
  const exercise = dummyExerciseData;

  if (exercise !== null) {
    const courseId = exercise.courseId || "defaultCourseId";
    const sectionId = exercise.sectionId || "defaultSectionId";

    if (exercise === true) {
        navigation.navigate("SectionComplete", {
        courseId: courseId,
        sectionId: sectionId,
      });
    } else {
      setExerciseData(exercise);
    }
  } else {
    navigation.navigate("ErrorScreen");
  }
}
*/

