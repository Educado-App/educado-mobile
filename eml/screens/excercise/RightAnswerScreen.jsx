import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { useNavigation, useRoute } from '@react-navigation/native';
import { Icon } from '@rneui/themed';
import { AppLoading } from 'expo-app-loading';

export default function RightAnswerScreen() {
  const navigation = useNavigation();

  const route = useRoute();
  const { courseId, sectionId } = route.params;
  return (
    <View
      style={{
        flex: 0.5,
        justifyContent: 'space-evenly',
        alignItems: 'center',
      }}
      className="bg-babyBlue flex-auto justify-evenly align-center"
    >
      <View style={{ flex: 1.5 }}>
        <View style={{ paddingTop: '40%', paddingBottom: '10%' }}>
          <Text
            style={{
              fontSize: 45
            }}
            className="text-gray-600"
          >
            {/* GOOD JOB! */}
            BOM TRABALHO!
          </Text>
        </View>
        <View style={{ justifyContent: 'space-evenly' }}>
          <Icon
            style={{
              shadowOpacity: 0.8,
              shadowRadius: 2,
              shadowOffset: { width: 0, height: 0 },
              shadowColor: '#ffd633',
              elevation: 10,
            }}
            size={200}
            name="star"
            type="material-community"
            color="#fac12f"
          />
        </View>
      </View>
      <View style={{ top: '5%', flex: 0.7 }}>
        <View
          style={[
            styles.nextArrow,
            styles.buttonShadow,
            { shadowColor: '#2db300' },
          ]}
        >
          <Icon
            size={70}
            name="chevron-right"
            type="material-community"
            color="white"
            onPress={() =>
              navigation.navigate('Exercise', {
                sectionId,
                courseId,
              })
            }
          />
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  button: {
    borderRadius: 15,
    width: 140,
    height: 110,
  },
  buttonShadow: {
    shadowOpacity: 0.6,
    shadowRadius: 10,
    shadowOffset: { width: 0, height: 0 },
  },
  nextArrow: {
    borderRadius: 15,
    backgroundColor: '#2db300',
    width: 300,
    height: 75,
    position: 'relative',
  },
});
