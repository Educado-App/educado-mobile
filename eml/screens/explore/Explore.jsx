import React from 'react';
import { useState, useEffect } from 'react';
import { View, ScrollView, RefreshControl, Pressable } from 'react-native';
import FilterNavBar from '../../components/explore/FilterNavBar';
import ExploreCard from '../../components/explore/ExploreCard';
import * as StorageService from '../../services/StorageService';
import { useNavigation } from '@react-navigation/native';
import BaseScreen from '../../components/general/BaseScreen';
import IconHeader from '../../components/general/IconHeader';
import { shouldUpdate, determineCategory } from '../../services/utilityFunctions';
import Text from '../../components/general/Text';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import NetworkStatusObserver from '../../hooks/NetworkStatusObserver';

/**
 * Explore screen displays all courses and allows the user to filter them by category or search text.
 * @returns {JSX.Element} - Rendered component
 */
export default function Explore() {

	// Search text state
	const [searchText, setSearchText] = useState('');
	// Selected category state
	const [selectedCategory, setSelectedCategory] = useState(null);

	//Sets dummy data for courses (will be replaced with data from backend)
	const [courses, setCourses] = useState([]);
	const [subCourses, setSubCourses] = useState([]);
	const [isOnline, setIsOnline] = useState(false);
	const [refreshing, setRefreshing] = useState(false);
	const navigation = useNavigation();

	/**
  * Asynchronous function that loads the subscribed courses from storage and updates the state.
  * @returns {void}
  */
	async function loadSubscriptions() {
		const subData = await StorageService.getSubCourseList();
		if (shouldUpdate(subCourses, subData)) {
			if (subData.length !== 0 && Array.isArray(subData)) {
				setSubCourses(subData);
			}
			else {
				setSubCourses([]);
			}
		}
	}

	/**
  * Asynchronous function that loads the courses from storage and updates the state.
  * @returns {void}
  */
	async function loadCourses() {
		const courseData = await StorageService.getCourseList();
		if (shouldUpdate(courses, courseData)) {
			if (courseData.length !== 0 && Array.isArray(courseData)) {
				setCourses(courseData);
			}
			else {
				setCourses([]);
			}
		}
	}

	// When refreshing the loadCourse and load subscription function is called
	const onRefresh = () => {
		setRefreshing(true);
		loadSubscriptions();
		loadCourses();
		setRefreshing(false);
	};

	useEffect(() => {
		// this makes sure loadcourses is called when the screen is focused
		const update = navigation.addListener('focus', () => {
			loadCourses();
			loadSubscriptions();
		});
		return update;

	}, [navigation, subCourses, selectedCategory, searchText, isOnline]);

	const checkIfSubscribed = (course, subCourses) => {
		for(let subCourse of subCourses){
			if (subCourse.courseId === course.courseId){
				return true;
			}
		}
		return false;
	};

	///---------------------------------------------///

	// Function to filter courses based on searchText or selectedCategory

	const filteredCourses = courses.filter((course) => {
		// Check if the course title includes the search text
		const titleMatchesSearch = course.title.toLowerCase().includes(searchText.toLowerCase());
		// Check if the course category matches the selected category (or no category is selected)
		const categoryMatchesFilter = !selectedCategory || determineCategory(course.category) === selectedCategory;
		// Return true if both title and category conditions are met
		return titleMatchesSearch && categoryMatchesFilter;
	});

	const handleFilter = (text) => {
		setSearchText(text);
		// console.log("handleFilter", searchText);
	};

	const handleCategoryFilter = (category) => {
		//if category label is "all" it will display all courses, otherwise it will display courses with the selected category
		if (category === 'Todos') {
			setSelectedCategory(null); // Set selectedCategory to null to show all items
		} else {
			setSelectedCategory(category); // Set selectedCategory to the selected category label
		}
	};

	return (
		<>
			<NetworkStatusObserver setIsOnline={setIsOnline} />
			<BaseScreen>
				<IconHeader
					title={'Explorar cursos'}
					description={'Inscreva-se nos cursos do seu interesse e comece sua jornada'}
				/>
				{!isOnline ?
					<View>
						<View className="justify-center px-1 pt-6">
							<MaterialCommunityIcons name="wifi-off" size={160} color="black" style={{ alignSelf: 'center' }} />
							<Text className="text-center font-montserrat-semi-bold text-[24px]">
								{'\n'}Sem conexão com internet.
							</Text>
							<View className="flex-row flex-wrap justify-center">
								<Text className="text-center text-body">
									{/* You are offline. Connect to the internet to explore the courses. */}
									{'\n'}Você está sem acesso a internet. Vá para
								</Text>
								<View className="flex-row flex-wrap justify-center">
									<Text className="text-center text-body font-montserrat-bold">
                    meus cursos
									</Text>
									<Text className="text-center text-body">
                    e acesse os cursos baixados.{'\n'}
									</Text>
								</View>
							</View>
							<View className="items-center pt-6">
								<Pressable
									testID={'offlineExploreButton'}
									className="rounded-r-8 rounded-md bg-primary justify-center items-center p-2 h-14 w-80"
									onPress={() => navigation.navigate('Meus cursos')}>
									{/* Click to explore courses */}
									<Text className="text-projectWhite font-sans-bold text-center text-body" >Ir para Meus cursos</Text>
								</Pressable>
							</View>
						</View>
					</View>
					:
					<View height="77%">
						<FilterNavBar
							onChangeText={(text) => handleFilter(text)}
							onCategoryChange={handleCategoryFilter}
						/>
						<ScrollView showsVerticalScrollIndicator={false} refreshControl={<RefreshControl refreshing={refreshing} onRefresh={onRefresh} />}>
							<View className="overflow-y-auto">
								{courses && filteredCourses && filteredCourses.map((course, index) => (
									<ExploreCard
										key={index}
										isPublished={course.status === 'published'}
										subscribed={/*isSubscribed[index]*/checkIfSubscribed(course, subCourses)}
										course={course}
									></ExploreCard>
								))}
							</View>
						</ScrollView>
					</View>
				}
			</BaseScreen>
		</>
	);
}