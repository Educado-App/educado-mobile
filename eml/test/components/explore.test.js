import React from "react"
import { render, screen } from "@testing-library/react-native"
import Explore from "../../screens/explore/Explore"
import ExerciseScreen from "../../screens/excercise/ExerciseScreen"
import { expect, test } from "@jest/globals"
import { useNavigation } from "@react-navigation/native"
import { useRoute } from "@react-navigation/native"
import RightAnswerScreen from "../../screens/excercise/RightAnswerScreen"
import WrongAnswerComponent from "../../screens/excercise/WrongAnswerScreen"
import CourseScreen from "../../screens/courses/CourseScreen"
import sum from "./sum"

import { Animated } from 'react-native';

jest.useRealTimers()
test('Summing two values', () => {
    expect(sum(1, 2)).toBe(3);
})
const mockPlatform = OS => {
    jest.resetModules();
    jest.doMock("react-native/Libraries/Utilities/Platform", () => ({ OS, select: objs => objs[OS] }));
};
it("my test on Android", () => {
    mockPlatform("android");
});
test('Testing if Explore renders', () => {
    render(<Explore />);
    expect(screen.toJSON()).toMatchSnapshot();
});
// mock useNavigate
const mockedUsedNavigate = jest.fn();
const mockedUseRoute = jest.fn();
jest.mock('@react-navigation/native', () => ({
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => mockedUsedNavigate,
    useRoute: () => mockedUseRoute,
    useParams: () => ({
        courseId: 0,
        sectionId: 0,
    }),
}))

// Doesn't work cause route.params
// test('Testing if RightAnswerScreen render', () => {
//     render(<RightAnswerScreen courseId={0} />);
//     expect(screen.toJSON()).toMatchSnapshot();
// })

// test('Testing if WrongAnswerScreen render', () => {
//     render(<WrongAnswerComponent />);
//     expect(screen.toJSON()).toMatchSnapshot();
// })

test('Testing if CourseScreen render', () => {
    render(<CourseScreen />);
    expect(screen.toJSON()).toMatchSnapshot();
})

// Doesn't work cause of route.params

// test('Testing if ExerciseScreen renders', () => {
//     render(<ExerciseScreen sectionId={0} />);
//     expect(screen.toJSON()).toMatchSnapshot();
// });
