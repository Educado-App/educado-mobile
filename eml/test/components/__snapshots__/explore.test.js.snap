// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing if CourseScreen render 1`] = `null`;

exports[`Testing if Explore renders 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#CFE9EF",
      },
      {
        "flexBasis": "100%",
      },
      {
        "display": "flex",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flexBasis": "16.666667%",
        },
        {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingTop": 20,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#4b5563",
          },
          {
            "fontFamily": "VarelaRound_400Regular",
            "fontSize": 40,
          },
        ]
      }
    >
       Educado
    </Text>
  </View>
  <View
    style={
      [
        {
          "width": "91.666667%",
        },
        {
          "alignSelf": "center",
        },
        {
          "paddingBottom": 16,
        },
        {
          "elevation": 15,
          "zIndex": 15,
        },
      ]
    }
  >
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderColor": "gray",
            "borderRadius": 10,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 12,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": undefined,
              },
              undefined,
            ]
          }
        >
          Filter by
        </Text>
        <Image
          resizeMode="contain"
          source={1}
          style={
            {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          [
            {
              "flexWrap": "wrap",
            },
            {
              "flexDirection": "row",
            },
            {
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
            },
            {
              "justifyContent": "space-evenly",
            },
          ]
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
